### ISSUE TITLE
[Bug]: Legal and tax classifications are guessed instead of blocked when facts are ambiguous

### ISSUE THREAD

**Issue by rahuldass19** [QWED-AI/qwed-tax#17]
## Summary

**Severity:** HIGH

Some `qwed-tax` classification paths resolve uncertainty by **choosing** a legal or tax classification instead of surfacing ambiguity. The system is not proving the classification — it is guessing one. This directly affects withholding, filing obligations, and tax rates.

---

## Problem

This appears in places where:

- Mixed worker-classification facts still collapse into a contractor/employee answer
- Unsupported or partially modeled asset types still get a capital-gains treatment
- Slab-dependent tax treatment is treated as verified without enough proof

In tax and compliance workflows, classification errors are not recoverable after the fact.

---

## Why this violates QWED philosophy

QWED requires:

- Ambiguous or partial cases must not return success
- No heuristic classification for legal or financial decisions
- Unproven results must never be treated as valid

If the system does not have enough facts to **prove** a classification, it must block or return an explicit ambiguous/unverifiable state.

---

## Affected scope

This pattern appears in classification-oriented logic:

- `qwed_tax/guards/classification_guard.py`
- `qwed_tax/guards/capital_gains_guard.py`

Representative problems:

- Worker classification defaults to contractor in mixed-signal cases
- Unknown asset types fall into a default holding-period threshold
- Slab-rate scenarios are treated as verified without actual rate proof

---

## Attack scenario

A caller provides incomplete or mixed facts but still gets a determinate legal/tax answer.

```python
verify_classification_claim(
    "1099",
    {
        "provides_tools": False,
        "reimburses_expenses": True,
        "indefinite_relationship": True
    }
)
```

This is not a conclusively safe contractor case, but the logic can still collapse toward contractor treatment.

```python
determine_term("2020-01-01", "2021-06-01", "unlisted_shares")
```

If the asset type is not explicitly modeled, a default threshold is applied and a tax treatment is still returned.

```python
verify_tax_rate("debt", "STCG", "0%")
```

The slab-rate path does not truly verify the claimed rate, but still returns a success-like result.

---

## Security impact

This can lead to:

- Wrong W-2 vs 1099 treatment
- Wrong capital-gains classification
- Wrong tax-rate acceptance
- Legally meaningful decisions derived from incomplete facts

In production, a user or agent can receive a confident-looking answer where the system only had partial information.

---

## Fix direction

- Introduce explicit `AMBIGUOUS`, `UNVERIFIABLE`, or `BLOCKED` outcomes for mixed or unsupported cases
- Remove default classification fallbacks for unsupported asset or fact combinations
- Require complete fact coverage before returning worker/tax classifications
- Treat slab-dependent outcomes as unresolved unless the actual applicable rate is proven
- Separate "computed guess" from "verified classification" at the API/result level

---

## Goal

`qwed-tax` should never guess a legal or tax classification.

If the facts do not conclusively prove the classification, the result must be **blocked or marked unverifiable**.

**Comment by rahuldass19** [comment]
## Audit findings — exact locations + 3 additional guards

Ran a full fail-closed audit. This issue's scope is confirmed with exact line numbers, plus 3 additional guards exhibiting the same "guess instead of block" pattern.

### Findings already in this issue (with line numbers)

| Guard | File | Lines | Exact behavior |
|-------|------|-------|----------------|
| CapitalGainsGuard (SLAB) | `guards/capital_gains_guard.py` | 57-60 | `if expected == "SLAB": return {"verified": True, "note": "Subject to Slab Rates"}` — `claimed_rate` is **never inspected**. Comment says "we accept it if LLM didn't claim a fixed low rate" but code doesn't check. Any rate on debt/LTCG/STCG passes. |
| CapitalGainsGuard (threshold) | `guards/capital_gains_guard.py` | 33 | `limit = thresholds.get(asset_type.lower(), 1095)` — unknown asset types get a fabricated 1095-day threshold, then flow into C3's `verified=True` pass. |

### Additional guards not listed in this issue

| Guard | File | Lines | Behavior | Severity |
|-------|------|-------|----------|----------|
| SpeculationGuard | `guards/speculation_guard.py` | 27-44 | Classification via substring: `is_speculative_loss = "intraday" in loss_source`. Any `loss_source` without "intraday" substring (e.g. `"options"`, `"futures"`, typo `"intra day"`) → classified non-speculative → set-off allowed. Unknown source = guessed benign. | MEDIUM |
| GSTGuard (RCM) | `jurisdictions/india/guards/gst_guard.py` | 110-112, 140-148 | Unknown service coerced to `OTHER`, unknown entity coerced to `INDIVIDUAL` via `_coerce()`. Coercion = guessing classification. Can suppress RCM liability for `DIRECTOR`/`SPONSORSHIP` services when recipient is unrecognized. | HIGH |
| SetoffGuard | `jurisdictions/india/guards/setoff_guard.py` | 64-67 | Comment: `"# If no restriction found, it's allowed (Default Allow)"`. Only 4 loss heads have restrictions; `OTHER_SOURCES` and any future head default to allowed. Unknown head = guessed allowed. | MEDIUM |

### Test enforces the fail-open (GSTGuard)

**`test_rcm.py` lines 145-148** — `test_unknown_string_service_is_forward_charge` asserts that `"MYSTERY"` service → `is_rcm=False`. This test **enforces the insecure behavior** — fixing H3 requires updating this test to assert fail-closed.

```python
# Current test (enforces fail-open)
def test_unknown_string_service_is_forward_charge(self):
    result = self.gst.verify_rcm_applicability("MYSTERY", "BODY_CORPORATE", "BODY_CORPORATE")
    assert result["is_rcm"] is False  # ← this must change to True/UNVERIFIABLE
```

### Related: #18 (success without comparing claim)

**`jurisdictions/india/guards/crypto_guard.py` lines 61-62** — `if vda_income <= 0: return TaxResult(verified=True, message="No VDA Income")` — returns `verified=True` without inspecting `claimed_tax`. Also treats negative `vda_income` (a loss) as "no income" instead of routing to set-off/lapse logic. This is #18's pattern (success without comparing claim to computed truth) intersecting with #17's pattern (guessing classification of zero/negative income).

### Recommended fix priority

1. **H5 (SLAB rate)** — HIGH, any claimed rate on debt passes unconditionally
2. **H3+H4 (GST RCM coercion)** — HIGH, unknown entity suppresses statutory RCM (update test too)
3. **M4 (capital gains threshold)** — MEDIUM, fabricated threshold for unknown assets
4. **M2 (speculation substring)** — MEDIUM, heuristic classification from substring match
5. **M1 (setoff default allow)** — MEDIUM, unknown loss head defaults to allowed
